src/bls/bls.cpp

```diff
@@ -124,9 +124,13 @@ CBLSSignature CBLSSecretKey::Sign(const uint256& hash) const
     }
 
     CBLSSignature sigRet;
-    sigRet.impl = Scheme(bls::bls_legacy_scheme.load())->Sign(impl, bls::Bytes(hash.begin(), hash.size()));
+    try {
+        sigRet.impl = Scheme(bls::bls_legacy_scheme.load())->Sign(impl, bls::Bytes(hash.begin(), hash.size()));
+        sigRet.fValid = true;
+    } catch (...) {
+        sigRet.fValid = false;
```

Comment:
I've thought about this more and I think instead of capturing every type of exception and just marking it as failed, we should have the exception logged so the cause of failure can be attributed appropriately, allowing us to distinguish between faulty input, faulty logic or sporadic known bugs (this being our case).

I think the program should still hard crash if an exception is raised that we aren't already aware of and excluded as "log only and try again".

Reply:
@UdjinM6 thoughts?

Reply:
Logging caught exceptions - sure, why not, but crashing on unknown exceptions - no, I don't think it's the right place for such behaviour. 1. it's too dependent on external input we can't control 2. we shouldn't crash just because the lib we use thrown an exception we don't understand (yet). "Mark it invalid (and drop it)" is the right way imo. I'd rather see a failed DKG or a failed IS/CL signing attempt than have the whole network (or even large portions of it) go down.

Reply:
That's understandable, I was hoping these hard crashes would happen during our internal testnets so that they're caught and documented, if not repaired alright but the possibility of a hard crash on mainnet could leave a possible DoS vector open if the library throws an exception at intentionally malformed input that isn't caught.

So, taking that point in mind, it seems that logging but not crashing would be the best approach for now. I think the linter can be asked to ignore certain files or error types. Thanks for your inputs @ogabrielides  and @UdjinM6!